source/agora/consensus/EnrollmentManager.d

```diff
@@ -190,6 +190,34 @@ public class EnrollmentManager
         return this.validator_set.getEnrolledHeight(enroll_hash);
     }
 
+    /***************************************************************************
+
+        Add a enrollment data to the enrollment pool
+
+        Params:
+            enroll = the enrollment data to add
+            block_height = current block height
+            finder = the delegate to find UTXOs with
+
+        Returns:
+            true if the enrollment data has been added to the enrollment pool
+
+    ***************************************************************************/
+
+    public bool addEnrollment (const ref Enrollment enroll, Height block_height,
+        scope UTXOFinder finder) @safe nothrow
+    {
+        auto enrolled_height = this.getEnrolledHeight(enroll.utxo_key);
+        ulong avail_height;
+
+        if (enrolled_height == ulong.max)
+            avail_height = block_height + 1 + this.params.ValidatorCycle;
```

Comment:
I don't understand this part. Why do you set `avail_height` to 1009 blocks in the future when there is no Enrollment ?

Reply:
The `block_height` + 1 means the enrolled height if the enrollment has been enrolled at the next block of the 'block_height`. So the `avail_height` becomes the last height at which a node plays a role as a validator. This assignment for the `avail_height` is only relevant to nodes requesting enrollment after the Genesis block.